mm2src/coins/coin_errors.rs

```diff
@@ -0,0 +1,82 @@
+use crate::{utxo::rpc_clients::UtxoRpcError, DelegationError, NumConversError, TxHistoryError,
+            UnexpectedDerivationMethod, WithdrawError};
+use futures01::Future;
+use mm2_err_handle::prelude::MmError;
+use spv_validation::helpers_validation::SPVError;
+
+pub type ValidatePaymentFut<T> = Box<dyn Future<Item = T, Error = MmError<ValidatePaymentError>> + Send>;
+
+#[derive(Debug, Display)]
+pub enum ValidatePaymentError {
+    InternalError(String),
+    // Problem with deserializing the transaction, or one of the transaction parts is invalid.
+    InvalidPaymentTxData(String),
+    InvalidInput(String),
+    InvalidRpcResponse(String),
+    SPVError(SPVError),
+    UnexpectedPaymentState(String),
+    Transport(String),
+    // Transaction has wrong properties, for example, it has been sent to a wrong address
+    WrongPaymentTx(String),
+    ValidateHtlcError(String),
```

Comment:
Is this error variant used? If yes, this can be replaced with `WrongPaymentTx` or `InvalidPaymentTxData`